include/deal.II/grid/tria.h

```diff
@@ -511,6 +513,253 @@ namespace internal
 /*------------------------------------------------------------------------*/
 
 
+namespace parallel
+{
+  namespace fullydistributed
+  {
+    /**
+     * Configuration flags for fully distributed Triangulations.
+     * Settings can be combined using bitwise OR.
+     *
+     * @author Peter Munch, 2019
+     */
+    enum Settings
+    {
+      /**
+       * Default settings, other options are disabled.
+       */
+      default_setting = 0x0,
+      /**
+       * This flags needs to be set to use the geometric multigrid
+       * functionality. This option requires additional computation and
+       * communication.
+       */
+      construct_multigrid_hierarchy = 0x1
+    };
+
+    /**
+     * Information needed for each locally relevant cell, stored in
+     * ConstructionData and used during construction of a
+     * parallel::fullydistributed::Triangulation. This struct stores
+     * the cell id, the subdomain_id and the level_subdomain_id as well as
+     * information related to manifold_id and boundary_id.
+     *
+     * @author Peter Munch, 2019
+     */
+    template <int dim>
+    struct CellData
+    {
+      /**
+       * Constructor.
+       */
+      CellData() = default;
+
+      /**
+       * Boost serialization function
+       */
+      template <class Archive>
+      void
+      serialize(Archive &ar, const unsigned int /*version*/);
+
+      /**
+       * Comparison operator.
+       */
+      bool
+      operator==(const CellData<dim> &other) const;
+
+      /**
+       * Unique CellID of the cell.
+       */
+      CellId::binary_type id;
+
+      /**
+       * subdomain_id of the cell.
+       */
+      types::subdomain_id subdomain_id;
+
+      /**
+       * level_subdomain_id of the cell.
+       */
+      types::subdomain_id level_subdomain_id;
+
+      /**
+       * Manifold id of the cell.
+       */
+      types::material_id manifold_id;
+
+      /**
+       * Manifold id of all lines of the cell.
+       *
+       * @note Only used for 2D and 3D.
+       */
+      std::array<types::manifold_id, GeometryInfo<dim>::lines_per_cell>
+        manifold_line_ids;
```

Comment:
This is better than the original version of CellData. My vote would be to simply use this one instead of the old one, with a constructor that does "the right thing" when initialized in the same way as it was before.